core/dom/iterator.js

```diff
@@ -392,6 +394,28 @@
 		}
 	};
 
+	/**
+	 * @param {CKEDITOR.dom.element} element
+	 * @returns {boolean} whether an element has nested non-text elements
+	 */
+	function hasNestedElements(element) {
+		if (!(element instanceof CKEDITOR.dom.element)) {
+			return false;
+		}
+		var children = element.getChildren();
+		if (!children) {
+			return false;
+		}
+		var count = children.count();
+		for (var i = 0; i < count; i++) {
```

Comment:
trying to be careful about adding expensive functions to the iterator since it's used by everything in ckeditor, 
practically this loop shouldn't need to iterate more than once to either find an element or reach the end of the list